implementations/rust/channel/src/lib.rs

```diff
@@ -317,7 +317,8 @@ impl<I: KeyExchanger, R: KeyExchanger, E: NewKeyExchanger<I, R>> ChannelManager<
 
         m.onward_route.addresses.remove(0);
 
-        let mut channel = Channel::new(channel_id, Box::new(self.new_key_exchanger.initiator()));
+        let mut channel =
+            Channel::new(channel_id, Box::new(self.new_key_exchanger.initiator(None)));
```

Comment:
Within `fn initiate_key_exchange`, is the expectation that both parties have no intention of using pre-existing keys? 

e.g. this function is _not_ called in the case that I do want to supply keys?

Reply:
I'm assuming @robinbudd will take care of that part. I just put this fix here so the tests pass.

Reply:
Ah, right -- sorry, that's correct as far as I am aware. thanks for calling it out.